src/pages/Market/Market.tsx

```diff
@@ -1,12 +1,84 @@
-import React from "react";
+import React, { createContext, useContext } from "react";
 
-import { Link } from "react-router-dom";
+import { Option } from "@components/Dropdown/Dropdown";
+import { CURRENCIES } from "@config/currencies";
+
+import CoinFilter from "./components/CoinFilter/CoinFilter";
+import CoinList from "./components/CoinList/CoinList";
+import CurrencyFilter from "./components/CurrencyFilter/CurrencyFilter";
+import MarketChange from "./components/MarketChange/MarketChange";
+import MarketSearch from "./components/MarketSearch/MarketSearch";
+import styleMarket from "./Market.module.scss";
+
+const OpenSearch = createContext({
+  openSearch: false,
+  setSearch: (value: boolean) => {},
+});
+const OpenSearchProvider = OpenSearch.Provider;
+export const useOpenSearchContext = () => useContext(OpenSearch);
+
+type MarketProps = {
+  onCurrencyChange: (value: Option) => void;
+};
+
+const Market: React.FC<MarketProps> = ({ onCurrencyChange }) => {
+  const [openSearch, setSearch] = React.useState<boolean>(false);
```

Comment:
чтобы было более понятно:
```suggestion
  const [openSearch, setOpenSearch] = React.useState<boolean>(false);
```

а ещё не нужно generic-тип указывать